idl/internal/docstring.go

```diff
@@ -0,0 +1,138 @@
+package internal
+
+import (
+	"strings"
+	"unicode"
+)
+
+const (
+	_docstringOpen  = "/**"
+	_docstringClose = "*/"
+)
+
+// CleanupDocstring takes a docstring in the form,
+//
+//  /**
+//   * foo bar
+//   */
+//
+// And returns,
+//
+//  foo bar
+func CleanupDocstring(s string) string {
+	lines := strings.Split(s, "\n")
+	dedent(lines, true /* skipFirstIfUnindented */)
+
+	switch len(lines) {
+	case 0:
+		return s
```

Comment:
don't think this case can happen,
https://play.golang.org/p/fswgcibGYQ

Reply:
That's probably true but I would like to keep the check in for the sake of defensive programming.